hadoop-hdds/common/src/main/resources/ozone-default.xml

```diff
@@ -3040,7 +3039,14 @@
       OM/SCM/DN/S3GATEWAY Server connection timeout in milliseconds.
     </description>
   </property>
-
+  <property>
+    <name>ozone.om.grpc.maximum.response.length</name>
+    <value>134217728</value>
```

Comment:
Have we validated that list keys will never violate this max length (if I understand this correctly)?
How is this value selected?

Reply:
Thanks @kerneltime for reviewing this and for your comments.
This config defines the maximum message size limit that we send / receive over the gRPC channel.  
The value was selected based on the max size defined for Hadoop RPC messages :`ipc.maximum.response.length` with default size 128 * 1024 * 1024.

If a list keys message exceeds this maximum limit for the channel, then  it will not be processed and the RPC will fail with `RESOURCE_EXHAUSTED`.  The exception should be raised from the channel and propagated to the user client.

Reply:
Is this enforced at the client side or server side. As in will the server hit this when returning more than the max number of bytes?

Reply:
This is enforced by the both the client and server as we define the limit, max message size.  In the case of listing, the server will feed the payload within the` OMResponse` and the length of the message is enforced by the client in this case.  It raises a `RESOURCE_EXHAUSTED - message length exceed` io.grpc exception.

Thanks @kerneltime, in looking into handling this exception I found that it needs to be isolated by the failover proxy provider `retryPolicy` to _fail_ (not retry) on calling the `shouldFailover` method.  The failover provider and `retryPolicy shouldFailover` is currently refactored and will be included in HDDS-6433 that will be commited to the Master after the feature branch merge.